src/main/java/io/github/lukehutch/fastclasspathscanner/FastClasspathScanner.java

```diff
@@ -219,32 +217,22 @@ public synchronized static final String getVersion() {
 
     // -------------------------------------------------------------------------------------------------------------
 
-    private final static AtomicInteger threadIdx = new AtomicInteger();
+    private final static NamedThreadFactory threadFactory = new NamedThreadFactory("FastClasspathScanner-worker-", true);
 
     /** A ThreadPoolExecutor that can be used in a try-with-resources block. */
-    private class AutocloseableExecutorService extends ThreadPoolExecutor implements AutoCloseable {
-        public AutocloseableExecutorService(final int numThreads) {
+    private class AutoCloseableExecutorService extends ThreadPoolExecutor implements AutoCloseable {
+        public AutoCloseableExecutorService(final int numThreads) {
             super(numThreads, numThreads, 0L, TimeUnit.MILLISECONDS,
                     // FIFO work queue
                     new LinkedBlockingQueue<Runnable>(),
-                    // Thread factory that names threads and sets daemon status to true
-                    new ThreadFactory() {
-                        @Override
-                        public Thread newThread(final Runnable r) {
-                            final Thread t = new Thread(r,
-                                    "FastClasspathScanner-worker-" + threadIdx.getAndIncrement());
-                            // Kill worker threads if main thread dies
-                            t.setDaemon(true);
-                            return t;
-                        }
-                    });
+                    threadFactory);
         }
 
         /** Shut down thread pool on close(). */
         @Override
         public void close() {
             try {
-                shutdown();
+                shutdownNow();
```

Comment:
from what I could see all operations inside the try catch for AutoCloseableExecutorService are blocking so this should be safe?